connections/src/main/java/com/stripe/android/connections/ConnectionsSheetContract.kt

```diff
@@ -55,10 +64,32 @@ internal class ConnectionsSheetContract :
         }
     }
 
-    @Parcelize
-    data class Result(
-        val connectionsSheetResult: ConnectionsSheetResult
-    ) : Parcelable {
+    internal sealed class Result : Parcelable {
+        /**
+         * The customer completed the connections session.
+         * @param linkAccountSession The link account session connected
+         */
+        @Parcelize
+        data class Completed(
+            val linkAccountSession: LinkAccountSession,
+            val token: Token? = null
+        ) : Result()
+
+        /**
+         * The customer canceled the connections session attempt.
+         */
+        @Parcelize
+        object Canceled : Result()
+
+        /**
+         * The connections session attempt failed.
+         * @param error The error encountered by the customer.
+         */
+        @Parcelize
+        data class Failed(
+            val error: Throwable
+        ) : Result()
+
```

Comment:
We'll use this internal result class for `ConnectionSheet -> ConnectionSheetActivity` communication, and we'll transform it to the corresponding exposed `Result` classes:

- (new) `ConnectionsSheetForTokenResult` when `ConnectionSheet.createForToken`
- (unchanged) `ConnectionsSheetResult` when `ConnectionSheet.create`